dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -1504,29 +1430,28 @@ private void compactBuffer()
     @Override
     public int available() throws IOException
     {
-        // Do the check for closed first here to avoid data races
+        int availBytes = 0;
+        synchronized (readBuffer)
+        {
+            availBytes = readBuffer.getBytesAvailable();
+        }
+
         if (this.closed.get())
         {
             if (this.prefetchException != null)
             {
                 throw new IOException("Prefetch thread exited early exception:" + prefetchException.getMessage(), this.prefetchException);
             }
-
-            int bufferLen = this.readBufferDataLen.get();
-            int availBytes = bufferLen - this.readPos;
+            
             if (availBytes == 0)
             {
                 String prefix = "RowServiceInputStream.available(), file "   + dataPart.getFileName() + " part " + dataPart.getThisPart() + " on IP " + getIP() + ":";
 
-                // this.bufferWriteMutex.release();
-                IOException wrappedException = new IOException(prefix + "End of input stream, bufferLen:" + bufferLen + ", this.readPos:" + this.readPos + ", availableBytes=0");
+                IOException wrappedException = new IOException(prefix + "End of input stream");
```

Comment:
Is this enough information for debugger to understand the problem and how to fix?